### ISSUE TITLE
Have Solution to Inlines 

### ISSUE THREAD

**Issue by thegiffman** [digi604/django-smart-selects#136]
Hi there,

I show inlines as broken in the latest build - the chained menu does not populate correctly like it does in a normal admin. I've fixed the issue myself in the JavaScript file locally and things seem to work fine. My guess is that inline functionality was added a while ago, but was since broken and nobody noticed.

Here's the code change to **chainedfk.js**:
Old:

```
                $(chainfield).change(function(){
                    var start_value = $(id).val();
                    var val = $(this).val();
                    fill_field(val, start_value, id, url, empty_label, auto_choose);
                })
```

New:

```
                $(chainfield).change(function(){
                    // Handle the case of inlines, where the ID will depend on which list item we are dealing with
                    var localID = id;
                    if (localID.indexOf("__prefix__") > -1) {
                        var prefix = $(this).attr("id").match(/\d+/)[0];
                        localID = localID.replace("__prefix__", prefix);
                    }

                    var start_value = $(localID).val();
                    var val = $(this).val();
                    fill_field(val, start_value, localID, url, empty_label, auto_choose);
                })
```

Please excuse my ignorance, but this is my first attempt at contributing to GitHub. I'd like to submit my fix for consideration for the community. How do I go about getting permission to do this?

**Comment by SalahAdDin** [comment]
Thank you very much man!
@digi604 Consider use this please.

**Comment by thegiffman** [comment]
No problem - thanks for this very handy addition to the admin. Oh, one more thing - for our purposes we found we really wanted to prepopulate chained foreign keys with the first value in the list. Most of our list items were sorted by date (latest submissions of a software project) and the majority use case is to select the latest. So for our case we changed this test for the "auto_choose" case to ">="

```
                    if(auto_choose && j.length >= 1){
                        $(elem_id + ' option[value="'+ j[0].value +'"]').attr('selected', 'selected');
                    }
```

This may not be what you want as the default auto choose behavior, but it was what we wanted. FYI.

**Comment by SalahAdDin** [comment]
Can you make a pr?

**Comment by thegiffman** [comment]
Not sure. I think I need to push to a branch first, correct? And I don't think I have permission to do that - at least when I tried, it asked for master credentials (even though I was pushing to a new branch). Apologies - I've never submitted to GitHub before.

**Comment by GoTop** [comment]
@thegiffman，I edit the js file as you said, but in inline, django-smart-selects
 still don't work.

**Comment by millin** [comment]
@GoTop , all works fine for me. What is your Django version?

**Comment by GoTop** [comment]
@millin 
I use Django 1.9.9

I install your fork‘s lastest commit with this command: 
pip install git+git://github.com/millin/django-smart-selects.git@9ba57a34767611f3c63ac7a24a985763d88c8c31

**Comment by millin** [comment]
@GoTop
And it not works forgot you?

**Comment by GoTop** [comment]
@millin 
yes , it doesn't work in inline


This is my model.py:
```
class Group(models.Model):
    name_cn = models.CharField(max_length=200, null=True, blank=True)
    name_en = models.CharField(max_length=200, null=True, blank=True)
    name_kr = models.CharField(max_length=200, null=True, blank=True)
    wiki_url = models.URLField(null=True, blank=True)

    def __unicode__(self):
        if self.name_cn:
            return self.name_cn

class Singer(models.Model):
    name_cn = models.CharField(max_length=200, null=True, blank=True)
    name_en = models.CharField(max_length=200, null=True, blank=True)
    name_kr = models.CharField(max_length=200, null=True, blank=True)
    group = models.ForeignKey('Group', null=True, blank=True)
    wiki_url = models.URLField(null=True, blank=True)

    def __unicode__(self):
        if self.name_cn:
            return self.name_cn

class Song(models.Model):
    name_cn = models.CharField(max_length=200, null=True, blank=True)
    name_en = models.CharField(max_length=200, null=True, blank=True)
    name_kr = models.CharField(max_length=200, null=True, blank=True)
    # member = models.ForeignKey('Singer', null=True, blank=True)
    group = models.ForeignKey('Group', null=True, blank=True)
    wiki_url = models.URLField(null=True, blank=True)

    member = ChainedForeignKey(
        Singer,
        chained_field="group",
        chained_model_field="group",
        show_all=False,
        auto_choose=True
    )
```


This is my install module:


```
pip list
Django (1.9.9)
django-debug-toolbar (1.5)
django-smart-selects (1.2.3)
futures (3.0.5)
google-api-python-client (1.5.3)
goslate (1.5.1)
httplib2 (0.9.2)
isodate (0.5.4)
oauth2client (3.0.0)
pip (8.1.2)
pyasn1 (0.1.9)
pyasn1-modules (0.0.8)
PySocks (1.5.7)
python-dateutil (2.5.3)
requests (2.11.1)
rsa (3.4.2)
setuptools (0.6rc11)
simplejson (3.8.2)
six (1.10.0)
sqlparse (0.2.1)
translate (2.0.0)
translation (1.0.5)
uritemplate (0.6)
win-inet-pton (1.0.1)
```

**Comment by minholi** [comment]
This is not working to me. I'm using django 1.10 with master branch (1.2.5)

**Comment by blag** [comment]
@minholi Thank you for posting your Django and D-S-S versions, but can you be a little more specific: *how* does it not work? And are there any errors logged in your browser console? Does Python dump a traceback? Thanks!

**Comment by blag** [comment]
This may be a duplicate of #60.

**Comment by minholi** [comment]
@blag sorry! There is no error on terminal or javascript console, it just returns a empty list [] when I open the chaining url that pops on the terminal when the change page is served. I'm using models and inlines just like @GoTop example and Python 3.5.2 (32bit) on windows.

**Comment by millin** [comment]
On master branch all works fine.

**Comment by blag** [comment]
I pushed changes in `master` out to PyPI with version 1.2.6. Can you upgrade to that version and test it again?

**Comment by blag** [comment]
Other very similar issues seem to have been solved with the 1.2.6 release, so I'm closing this. Feel free to reopen if it's not.

**Comment by Serebrennikov-Aleks** [comment]
Hi there,

it seems it doesn't work in inline (admin.TabularInline)

This is my model config:

```
class Deal(models.Model):
    name = models.CharField(max_length=100)
    description = models.TextField()
    price = models.FloatField(verbose_name='Base price', default=0)
    photo = models.ImageField(upload_to='uploads/avatar/', max_length=100)
    start_period = models.DateField()
    end_period = models.DateField()

    def __str__(self):
        return self.name


class DealComposition(models.Model):
    category = models.ForeignKey(Category, on_delete=models.CASCADE)
    deal = models.ForeignKey(Deal, on_delete=models.CASCADE)

    product = ChainedForeignKey(
        Product,
        chained_field="category",
        chained_model_field="category",
        show_all=False,
        auto_choose=False,
        sort=True
    )

    attribute = ChainedForeignKey(
        Attribute,
        show_all=False,
        auto_choose=False,
        sort=True,
        chained_field="product",
        chained_model_field="product"
    )

    option = ChainedManyToManyField(
        Option,
        horizontal=True,
        verbose_name='option',
        chained_field="attribute",
        chained_model_field="attribute"
    )


class DealCompositionTablesInline(admin.TabularInline):
    model = DealComposition
    extra = 0


class DealAdmin(admin.ModelAdmin):
    list_display = ['name', 'description', 'price', 'photo']

    inlines = [DealCompositionTablesInline]
```

And this is my installed module info:
```
attrs (17.4.0)
certifi (2018.1.18)
chardet (3.0.4)
click (6.7)
coreapi (2.3.3)
coreapi-cli (1.0.9)
coreschema (0.0.4)
decorator (4.2.1)
Django (2.0.3)
django-debug-toolbar (1.9.1)
django-nested-admin (3.0.21)
django-smart-selects (1.5.4)
djangorestframework (3.7.7)
idna (2.6)
ipdb (0.11)
ipython (6.2.1)
ipython-genutils (0.2.0)
itypes (1.1.0)
jedi (0.11.1)
Jinja2 (2.10)
MarkupSafe (1.0)
parso (0.1.1)
pexpect (4.4.0)
pickleshare (0.7.4)
Pillow (5.0.0)
pip (9.0.3)
pluggy (0.6.0)
prompt-toolkit (1.0.15)
psycopg2 (2.7.4)
psycopg2-binary (2.7.4)
ptyprocess (0.5.2)
py (1.5.3)
Pygments (2.2.0)
python-monkey-business (1.0.0)
pytz (2018.3)
requests (2.18.4)
setuptools (39.0.1)
simplegeneric (0.8.1)
six (1.11.0)
sqlparse (0.2.4)
traitlets (4.3.2)
uritemplate (3.0.0)
urllib3 (1.22)
wcwidth (0.1.7)
wheel (0.30.0)
```
